src/woo_publications/metadata/tests/test_theme_api_endpoints.py

```diff
@@ -62,7 +66,54 @@ def test_detail_theme(self):
             "uuid": str(theme.uuid),
             "identifier": "https://www.example.com/thema/1",
             "naam": "item one",
-            "depth": 1,
+            "subThemes": [],
         }
 
         self.assertEqual(data, expected_data)
+
+    def test_detail_theme_wrong_uuid(self):
+        list_url = reverse(
+            "api:theme-detail",
+            kwargs={"uuid": str(uuid.uuid4)},
+        )
+
+        response = self.client.get(list_url)
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_detail_broken_uuid(self):
+        # UUID misses 3 characters
+        list_url = reverse(
+            "api:theme-detail",
+            kwargs={"uuid": "d6323f56-5331-4b43-8e8c-63509be1e"},
+        )
+
+        response = self.client.get(list_url)
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_detail_normalize_tree_dump_errors_result_in_404(self):
+        list_url = reverse(
+            "api:theme-detail",
+            kwargs={"uuid": str(uuid.uuid4)},
+        )
+
+        with self.subTest("tree_dump_returns_dict_instead_of_object"):
+            with patch(
+                "woo_publications.metadata.api.viewsets._normalize_tree_dump",
+                return_value={},
+            ):
+
+                response = self.client.get(list_url)
+
+                self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+        with self.subTest("tree_dump_returns_empty_list"):
+            with patch(
+                "woo_publications.metadata.api.viewsets._normalize_tree_dump",
+                return_value=[{}],
+            ):
+
+                response = self.client.get(list_url)
+
+                self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```

Comment:
I don't understand why you'd break the implementation using unittest.mock and check the effect it has on the API endpoint.

If the implementation is broken, this should crash hard and report errors to Sentry, not "silently" fail with a 404 error.

Additionally, private functions (names defined with an underscore prefix) should *not* be tested in unit test, since they are private and an implementation detail by design. Tying tests and implementation details together removes the ability to refactor without having to touch tests.